src/Compilers/Core/Portable/CryptographicHashProvider.cs

```diff
@@ -206,7 +206,7 @@ internal static ImmutableArray<byte> ComputeHash(HashAlgorithmName algorithmName
         internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<byte> bytes, SourceHashAlgorithm hashAlgorithm = SourceHashAlgorithms.Default)
         {
             var algorithmName = GetAlgorithmName(hashAlgorithm);
-            using (var incrementalHash = IncrementalHash.CreateHash(algorithmName))
+            using (var incrementalHash = RoslynIncrementalHash.CreateHash(algorithmName))
             {
                 incrementalHash.AppendData(bytes.ToArray());
```

Comment:
yikes.  feels like this should be better on Core as well, by using available hash functions that don't require tehse intermediary allocations.